tools/run_tests/run_tests.py

```diff
@@ -217,6 +226,8 @@ def test_specs(self):
     return sorted(out)
 
   def make_targets(self):
+    if self.args.compiler is _FREE_PASS and self.args.travis:
```

Comment:
You've already tested for --travis in _compiler_choices, why are you testing for self.args.travis again here (and 10 other times elsewhere)?